internal/dnfjson/dnfjson.go

```diff
@@ -73,6 +77,42 @@ type RepoConfig struct {
 	MetadataExpire string `json:"metadata_expire,omitempty"`
 }
 
+// ReposFromRPMMD converts an rpmmd.RepoConfig to a RepoConfig. If the
+// repository requires a subscription, the system subscriptions are loaded and
+// included in the new configs.
+func ReposFromRPMMD(rpmRepos []rpmmd.RepoConfig, arch string, releaseVer string) ([]RepoConfig, error) {
+	subscriptions, _ := rhsm.LoadSystemSubscriptions()
```

Comment:
Does this mean we load subscriptions on every call, rather than once on start-up, or do I misremember the old code? Why is this method needed / added?

Reply:
Yes, subscriptions are loaded on every call.  Currently, in `main`, subscriptions are loaded at composer started (in `NewComposer()`), however, they're also loaded [every time the depsolve job is called ](https://github.com/osbuild/osbuild-composer/blob/2c2668f4933129bbedf20bc68fb8b727799a447c/cmd/osbuild-worker/jobimpl-depsolve.go#L22), because we need to initialise a new RPMMD in each worker.

This function (without the subscription loading) is the equivalent to [this translation function](https://github.com/osbuild/osbuild-composer/blob/2c2668f4933129bbedf20bc68fb8b727799a447c/internal/rpmmd/repository.go#L409) in current `main`).

The subscription loading can probably be moved to the `Solver` constructor.

Reply:
Would it be acceptable (for now) to load them once for every job (on prem)?  A single job will do multiple depsolves so those will reuse the subscription information for those calls, but reload on each request.